vidarr-pluginapi/src/main/java/ca/on/oicr/gsi/vidarr/BasicType.java

```diff
@@ -114,6 +114,23 @@
     public <R> R apply(Visitor<R> transformer) {
       return transformer.dictionary(key, value);
     }
+
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) {
+        return true;
+      }
+      if (o == null || getClass() != o.getClass()) {
+        return false;
+      }
+      DictionaryBasicType that = (DictionaryBasicType) o;
+      return key.equals(that.key) && value.equals(that.value);
```

Comment:
Not null safe - could use `Objects#equals`

Same for all other `equals` methods in this PR

Reply:
None of those values can be null. I'll add null checks to the constructors.

Reply:
Okay. Aggressive null checks.